dashboard/src/components/UpgradeForm/UpgradeForm.tsx

```diff
@@ -121,10 +121,10 @@ function UpgradeForm({
   }, [deployed.values, modifications]);
 
   useEffect(() => {
-    if (deployed.chartVersion?.attributes.version && !version) {
+    if (deployed.chartVersion?.attributes.version) {
```

Comment:
It seems to me we should either be clearing `version` in the state when changing routes, or ensuring that it is specific to the ID of the chart, rather than just re-requesting to ensure it's the right one. Sounds like a use-case for #865 ?

Reply:
Agree, the state can benefit from the refactor (but better fix this bug now)

Reply:
FWIW, it's not about whether the state can benefit from a refactor IMO, but whether we/contributors will save time in the long-term by being able to reason `is this chart already loaded in the state`, rather than creating a function, later finding a bug, updating with a cludge which re-loads the chart every time, etc. A little bit of house-keeping with each PR goes a long way.